### PR TITLE
New historical helmets

### PR DESCRIPTION
**Pull request by QuartzMech**
#### Summary
Content "Adds three new types of medieval helmet into the game."

#### Purpose of change

This change adds three new helmets to the list of existing medieval ones. The purpose of this is to expand the current limited list of western helmets, as well as to add helmets with new functionalities to the list of medieval armor. The three items added are the kettle helm, the armet helm, and the sallet helm.

#### Describe the solution

The three helmets added each have different functionality and fit into a different spot within the current roster.

- The kettle helm is a sturdy and protective helmet which only covers the head.
- The sallet helm is a full plate helmet with a visor over the eyes. This visor can be activated to move it up or down, in a similar function to the existing riot helmet.
- The armet helm is a full plate helmet that is similar in function and stats to the sallet, except that the armet has a full face visor and will cover the mouth and eyes when down. 

The recipes for these items have been added to the historical armorsmithing book, and are included in the medieval itemset. 

#### Describe alternatives you've considered

Alternatives include simply not adding more helmets or adding a different set of helmets.

#### Testing

All three items appear correctly in the medieval armorsmithing book and can be crafted at the appropriate skill levels. The visors can be raised and lowered as expected, and provide appropriate protection according to their status. 

#### Additional context
Here are some pictures of what each kind of helmet looks like
Sallet
image
Kettle
image

Armet
image

### CODE REVIEW

File: data/json/items/tool_armor.json [hunk 1]
```diff
@@ -2860,6 +2860,110 @@
...
+    "material_thickness": 4,
+    "environmental_protection": 1,
+    "techniques": [ "WBLOCK_1" ],
+    "flags": [ "VARSIZE", "STURDY" ]
+  },
+  {
+    "id": "helmet_sallet_raised",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str": "sallet helm (raised)" },
```
Comment (Qrox) [inline data/json/items/tool_armor.json:2893]:
```suggestion
    "name": { "str": "sallet helm (raised)", "str_pl": "sallet helms (raised)" },
```

File: data/json/items/tool_armor.json [hunk 2]
```diff
@@ -2860,6 +2860,110 @@
...
+    "material_thickness": 5,
+    "environmental_protection": 1,
+    "techniques": [ "WBLOCK_1" ],
+    "flags": [ "VARSIZE", "STURDY" ]
+  },
+  {
+    "id": "helmet_armet_raised",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str": "armet helm (raised)" },
```
Comment (Qrox) [inline data/json/items/tool_armor.json:2948]:
```suggestion
    "name": { "str": "armet helm (raised)", "str_pl": "armet helms (raised)" },
```

File: data/json/items/armor/helmets.json [hunk 3]
```diff
@@ -610,6 +610,37 @@
...
+    "weight": "2555 g",
+    "volume": "2050 ml",
+    "price": 30000,
+    "price_postapoc": 4000,
+    "bashing": 7,
+    "material": [ "steel", "leather" ],
+    "symbol": "n",
+    "looks_like": "helmet_barbute",
+    "color": "light_gray",
+    "armor": [ { "covers": [ "head" ], "coverage": 90, "encumbrance": 20 } ],
```
Comment (Maleclypse) [inline data/json/items/armor/helmets.json:628]:
```suggestion
    "armor": [ { "covers": [ "head" ], "coverage": 85, "encumbrance": 20 } ],
```

File: data/json/items/tool_armor.json [hunk 4]
```diff
@@ -2860,6 +2860,110 @@
...
+    "price_postapoc": 4500,
+    "use_action": { "type": "transform", "msg": "You raise the sallet visor.", "target": "helmet_sallet_raised", "menu_text": "Raise" },
+    "bashing": 8,
+    "material": [ "steel", "leather" ],
+    "symbol": "s",
+    "looks_like": "helmet_barbute",
+    "color": "light_gray",
+    "armor": [
+      { "covers": [ "head" ], "coverage": 100, "encumbrance": 20 },
+      { "covers": [ "eyes" ], "coverage": 100, "encumbrance": 20 }
```
Comment (Maleclypse) [inline data/json/items/tool_armor.json:2881]:
```suggestion
      { "covers": [ "eyes" ], "coverage": 60, "encumbrance": 20 }
```
100% coverage would be a clear transparent material over the eyes with no gaps.

File: data/json/items/tool_armor.json [hunk 5]
```diff
@@ -2860,6 +2860,110 @@
...
+    "price": 35000,
+    "price_postapoc": 4500,
+    "use_action": { "type": "transform", "msg": "You raise the sallet visor.", "target": "helmet_sallet_raised", "menu_text": "Raise" },
+    "bashing": 8,
+    "material": [ "steel", "leather" ],
+    "symbol": "s",
+    "looks_like": "helmet_barbute",
+    "color": "light_gray",
+    "armor": [
+      { "covers": [ "head" ], "coverage": 100, "encumbrance": 20 },
```
Comment (Maleclypse) [inline data/json/items/tool_armor.json:2880]:
```suggestion
      { "covers": [ "head" ], "coverage": 90, "encumbrance": 20 },
```

File: data/json/items/tool_armor.json [hunk 6]
```diff
@@ -2860,6 +2860,110 @@
...
+      "msg": "You lower the sallet visor over your eyes.",
+      "target": "helmet_sallet",
+      "menu_text": "Lower"
+    },
+    "bashing": 8,
+    "material": [ "steel", "leather" ],
+    "symbol": "s",
+    "looks_like": "helmet_barbute",
+    "color": "light_gray",
+    "armor": [ { "covers": [ "head" ], "coverage": 100, "encumbrance": 20 } ],
```
Comment (Maleclypse) [inline data/json/items/tool_armor.json:2910]:
```suggestion
    "armor": [ { "covers": [ "head" ], "coverage": 90, "encumbrance": 20 } ],
```

File: data/json/items/tool_armor.json [hunk 7]
```diff
@@ -2860,6 +2860,110 @@
...
+    "price_postapoc": 5500,
+    "use_action": { "type": "transform", "msg": "You raise the armet visor.", "target": "helmet_armet_raised", "menu_text": "Raise" },
+    "bashing": 8,
+    "material": [ "steel", "leather" ],
+    "symbol": "s",
+    "looks_like": "helmet_barbute",
+    "color": "light_gray",
+    "armor": [
+      { "covers": [ "head" ], "coverage": 100, "encumbrance": 40 },
+      { "covers": [ "eyes" ], "coverage": 100, "encumbrance": 20 },
```
Comment (Maleclypse) [inline data/json/items/tool_armor.json:2935]:
```suggestion
      { "covers": [ "eyes" ], "coverage": 70, "encumbrance": 20 },
```

File: data/json/items/tool_armor.json [hunk 8]
```diff
@@ -2860,6 +2860,110 @@
...
+      { "covers": [ "mouth" ], "coverage": 100, "encumbrance": 30 }
```
Comment (Maleclypse) [inline data/json/items/tool_armor.json:2936]:
```suggestion
      { "covers": [ "mouth" ], "coverage": 85, "encumbrance": 30 }
```

**Review by Maleclypse — CHANGES_REQUESTED** [review]
Sorry this has taken some time for me to get to.  I think these coverage values might be better.

File: data/json/items/tool_armor.json (hunk 8 above)
Comment (QuintusAquila) [reply data/json/items/tool_armor.json:2936]: 85 might be a bit low. I can't imagine anything outside a massive fluke striking someone through the breathing holes in the bevor/visor.

Comment (QuartzMech) [reply data/json/items/tool_armor.json:2936]: Pretty much that, I can see reducing it to like 95 or so, but going off the picture I posted there is almost no way to hit the person in the mouth while wearing an armet.

**Comment by QuintusAquila** [comment]
This is incredibly petty, but I'm about 95% sure the picture you used for the Armet is a close helmet. If you look closely the split in the bevor is on the side not the front.

File: data/json/items/tool_armor.json (hunk 8 above)
Comment (Maleclypse) [reply data/json/items/tool_armor.json:2936]: All good. I literally was going off Google pictures and trying to think of the percentage in relation to each other.  The percentage point differences don’t have to be by five points either way but I wanted some discussion

Comment (QuartzMech) [reply data/json/items/tool_armor.json]: yeah, its probably fine either way, I'll just commit the changes you suggested.

**Review by Maleclypse — APPROVED** [review]
Thanks for doing this!

**Pull request merged**